File: problems/1143/solution.py
class Solution:
    def longestCommonSubsequence(self, text1: str, text2: str) -> int:
        dp = [[0 for i in range(len(text1) + 1)] for j in range(len(text2) + 1)]
        
        for i in range(1, len(dp)):
            for j in range(1, len(dp[0])):
                dp[i][j] = max(dp[i - 1][j] ,dp[i][j - 1])
                
                if text2[i - 1] == text1[j - 1]:
                    dp[i][j] = dp[i - 1][j - 1] + 1 
                    
        return dp[-1][-1]

File: problems/1143/solution.cpp
#define ll long long int
typedef vector<ll> vll;
typedef vector<vll> vvll;

class Solution {
public:
    int longestCommonSubsequence(string text1, string text2) {
        ll n = text1.size(), m = text2.size();
        vvll dp(n + 1, vll(m + 1));
        
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= m; j++) {
                int dif = text1[i - 1] == text2[j - 1] ? 1 : 0;
                dp[i][j] = max({dp[i - 1][j], dp[i][j - 1], dp[i - 1][j - 1] + dif});
            }
        }
        
        return dp[n][m];
    }
};